### DISCUSSION TITLE
S1-U traffic is not going through GTP

### DISCUSSION THREAD

**Discussion by lanclinsj** [magma/magma#11542] [Q&A]
Hi Team, 

Here is our setup as below. NAT enabled mode is on. when UE attaches there is no issue reaching internet but there is no GTP in S1-U traffic. I could see the internet traffic on eth1 - for example when I ping the Public DNS 8.8.8.8 (attached the PCAP for reference) - is there any configuration mismatch because of that S1-U is not going over GTP. Please help me validating that. Attached the notes of MME log as well as IP details. 
[AGW_Attach_MME.log](https://github.com/magma/magma/files/8052366/AGW_Attach_MME.log)
[IP_Details.txt](https://github.com/magma/magma/files/8052367/IP_Details.txt)
[GTP-U_Check_Feb10_1622.pcap.zip](https://github.com/magma/magma/files/8052368/GTP-U_Check_Feb10_1622.pcap.zip)

+-------+  eth1 +-----+.   eth0 
|  eNB     | --->  | AGW | ----> Internet 
+-------+.         +-----+
 10.0.2.2  s1ap/s1u   10.0.2.1

**Comment by edaspb** [comment]
Hi! Do you see iptables nat rule?
`iptables -t nat -S`

**Comment by lanclinsj** [comment]
Hi

Here is the output of iptables

***@***.***:~# iptables -t nat -S
-P PREROUTING ACCEPT
-P INPUT ACCEPT
-P OUTPUT ACCEPT
-P POSTROUTING ACCEPT
-A PREROUTING -d 192.88.99.142/32 -p tcp -m tcp --dport 48080 -j DNAT --to-destination 10.0.2.1
-A POSTROUTING -o eth0 -j MASQUERADE
***@***.***:~#

***@***.***:~# iptables -t nat -S -vvv
-P PREROUTING ACCEPT -c 36318 3058772
-P INPUT ACCEPT -c 6783 1095884
-P OUTPUT ACCEPT -c 381115 25299144
-P POSTROUTING ACCEPT -c 23384 1392935
-A PREROUTING -d 192.88.99.142/32 -p tcp -m tcp --dport 48080 -c 0 0 -j DNAT --to-destination 10.0.2.1
-A POSTROUTING -o eth0 -c 387242 25865609 -j MASQUERADE
libiptc vlibxtables.so.12. 1192 bytes.
Table `nat'
Hooks: pre/in/fwd/out/post = 0/188/ffffffff/220/2b8
Underflows: pre/in/fwd/out/post = f0/188/ffffffff/220/360
Entry 0 (0):
SRC IP: 0.0.0.0/0.0.0.0
DST IP: 192.88.99.142/255.255.255.255
Interface: `'/................to `'/................
Protocol: 6
Flags: 00
Invflags: 00
Counters: 0 packets, 0 bytes
Cache: 00000000
Match name: `tcp'
Target name: `DNAT' [80]

Entry 1 (240):
SRC IP: 0.0.0.0/0.0.0.0
DST IP: 0.0.0.0/0.0.0.0
Interface: `'/................to `'/................
Protocol: 0
Flags: 00
Invflags: 00
Counters: 36318 packets, 3058772 bytes
Cache: 00000000
Target name: `' [40]
verdict=NF_ACCEPT

Entry 2 (392):
SRC IP: 0.0.0.0/0.0.0.0
DST IP: 0.0.0.0/0.0.0.0
Interface: `'/................to `'/................
Protocol: 0
Flags: 00
Invflags: 00
Counters: 6783 packets, 1095884 bytes
Cache: 00000000
Target name: `' [40]
verdict=NF_ACCEPT

Entry 3 (544):
SRC IP: 0.0.0.0/0.0.0.0
DST IP: 0.0.0.0/0.0.0.0
Interface: `'/................to `'/................
Protocol: 0
Flags: 00
Invflags: 00
Counters: 381115 packets, 25299144 bytes
Cache: 00000000
Target name: `' [40]
verdict=NF_ACCEPT

Entry 4 (696):
SRC IP: 0.0.0.0/0.0.0.0
DST IP: 0.0.0.0/0.0.0.0
Interface: `'/................to `eth0'/XXXXX...........
Protocol: 0
Flags: 00
Invflags: 00
Counters: 387242 packets, 25865609 bytes
Cache: 00000000
Target name: `MASQUERADE' [56]

Entry 5 (864):
SRC IP: 0.0.0.0/0.0.0.0
DST IP: 0.0.0.0/0.0.0.0
Interface: `'/................to `'/................
Protocol: 0
Flags: 00
Invflags: 00
Counters: 23384 packets, 1392935 bytes
Cache: 00000000
Target name: `' [40]
verdict=NF_ACCEPT

Entry 6 (1016):
SRC IP: 0.0.0.0/0.0.0.0
DST IP: 0.0.0.0/0.0.0.0
Interface: `'/................to `'/................
Protocol: 0
Flags: 00
Invflags: 00
Counters: 0 packets, 0 bytes
Cache: 00000000
Target name: `ERROR' [64]
error=`ERROR'

***@***.***:~#

From: edaspb ***@***.***>
Date: Tuesday, February 15, 2022 at 8:37 AM
To: magma/magma ***@***.***>
Cc: lanclinsj ***@***.***>, Author ***@***.***>
Subject: Re: [magma/magma] S1-U traffic is not going through GTP (Discussion #11542)

Hi! Do you see iptables nat rule?
iptables -t nat -S

—
Reply to this email directly, view it on GitHub<https://github.com/magma/magma/discussions/11542#discussioncomment-2182323>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXXH4HMMGS65CK4UAON3JPDU3J6LDANCNFSM5OF2R2HQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by edaspb** [reply]
As far as I get you are using 192.168.128.0/24 as a subscriber ip pool and the same network is attached to gtp_br0 interface.
I would suggest to try another pool for subscribers.

**Reply by lanclinsj** [reply]
any recommendation of IP pool or random is fine.

**Comment by edaspb** [comment]
Random is fine, anyone which would't interfere with agw OS routing table.

**Reply by lanclinsj** [reply]
I had made the changes of UE Pool as 192.168.52.0/24  in the Orc8r. restarted the AGW. UE got the IP from that pool. Still the same, I could not see GTP over S1-U. 

root@magmaagw:~#
root@magmaagw:~# ovs-vsctl show
d0a560e6-9797-4e72-835c-661c728a957c
    Manager "ptcp:6640"
    Bridge uplink_br0
        Port patch-agw
            Interface patch-agw
                type: patch
                options: {peer=patch-up}
        Port uplink_br0
            Interface uplink_br0
                type: internal
        Port dhcp0
            Interface dhcp0
                type: internal
    Bridge gtp_br0
        Controller "tcp:127.0.0.1:6654"
            is_connected: true
        Controller "tcp:127.0.0.1:6633"
            is_connected: true
        fail_mode: secure
        Port proxy_port
            Interface proxy_port
        Port ipfix0
            Interface ipfix0
                type: internal
        Port mtr0
            Interface mtr0
                type: internal
        Port patch-up
            Interface patch-up
                type: patch
                options: {peer=patch-agw}
        Port gtp0
            Interface gtp0
                type: gtpu
                options: {key=flow, remote_ip=flow}
        Port g_202000a
            Interface g_202000a
                type: gtpu
                options: {key=flow, remote_ip="10.0.2.2"}
        Port li_port
            Interface li_port
                type: internal
        Port gtp_br0
            Interface gtp_br0
                type: internal
    ovs_version: "2.15.2"
root@magmaagw:~# ip r
default via 10.1.10.1 dev eth0
10.0.2.0/24 dev eth1 proto kernel scope link src 10.0.2.1
10.1.0.0/24 dev mtr0 proto kernel scope link src 10.1.0.1
10.1.10.0/24 dev eth0 proto kernel scope link src 10.1.10.121
10.5.0.0/24 dev envoy_cntr proto kernel scope link src 10.5.0.2
127.0.0.0/24 dev ipfix0 proto kernel scope link src 127.0.0.10
127.1.0.0/24 dev li_port proto kernel scope link src 127.1.0.0
****
192.168.128.0/24 dev gtp_br0 proto kernel scope link src 192.168.128.1
root@magmaagw:~#


ESM message container contents: 5233c101090504696e65740501c0a8340c5e028686272780802110030000108106080808…
    0101 .... = EPS bearer identity: EPS bearer identity value 5 (5)
    .... 0010 = Protocol discriminator: EPS session management messages (0x2)
    Procedure transaction identity: 51
    NAS EPS session management messages: Activate default EPS bearer context request (0xc1)
    EPS quality of service
        Length: 1
        Quality of Service Class Identifier (QCI): QCI 9 (9)
    Access Point Name
        Length: 5
        APN: inet
    PDN address
        Length: 5
        0000 0... = Spare bit(s): 0x00
        PDN type: IPv4 (1)
        **PDN IPv4: 192.168.52.12**
    APN aggregate maximum bit rate
        Element ID: 0x5e
        Length: 2
        APN-AMBR for downlink: 960 kbps
        APN-AMBR for uplink: 960 kbps
    Protocol Configuration Options
        Element ID: 0x27
        Length: 39
        [Link direction: Network to MS (1)]
        1... .... = Extension: True
        .... .000 = Configuration Protocol: PPP for use with IP PDP type or IP PDN type (0)
        Protocol or Container ID: Internet Protocol Control Protocol (0x8021)
            Length: 0x10 (16)
            PPP IP Control Protocol
                Code: Configuration Nak (3)
                Identifier: 0 (0x00)
                Length: 16
                Options: (12 bytes), Primary DNS Server IP Address, Secondary DNS Server IP Address
                    Primary DNS Server IP Address
                        Type: Primary DNS Server IP Address (129)
                        Length: 6
                        Primary DNS Address: 8.8.8.8
                    Secondary DNS Server IP Address
                        Type: Secondary DNS Server IP Address (131)
                        Length: 6
                        Secondary DNS Address: 8.8.4.4
        Protocol or Container ID: DNS Server IPv4 Address (0x000d)
            Length: 0x04 (4)
            IPv4: 8.8.8.8
        Protocol or Container ID: P-CSCF IPv4 Address (0x000c)
            Length: 0x04 (4)
            IPv4: 172.27.23.150
        Protocol or Container ID: IPv4 Link MTU (0x0010)
            Length: 0x02 (2)
            IPv4 link MTU size: 1400 octets

**Reply by lanclinsj** [reply]
![Screen Shot 2022-02-15 at 12 46 10 PM](https://user-images.githubusercontent.com/99515933/154146387-69c576fb-8290-413d-8d72-9e9861a00c7e.png)

**Comment by pshelar** [comment]
@edaspb here is doc to debug datapath issues: https://docs.magmacore.org/docs/next/howtos/troubleshooting/datapath_connectivity

**Reply by lanclinsj** [reply]
thanks Pravin for sharing this link - there is no issue with UE browsing but there is no GTP messages. User plane is not going through GTP. I see all UE traffic without encap eth1 which connection between eNB --> AGW.

**Comment by lanclinsj** [comment]
Good to close this discussion - it was problem with Baicells radio - I had replaced the eNB unit after that there is no issue with GTP messages. Thanks for your help!
